src/dataset/iter.rs

```diff
@@ -2,6 +2,7 @@ use super::{AsTargets, DatasetBase, DatasetView, FromTargetArray, Records, Targe
 use ndarray::{ArrayBase, ArrayView, ArrayView1, ArrayView2, Axis, Data, Ix2};
 use std::marker::PhantomData;
 
+#[derive(Clone)]
```

Comment:
Why no `Debug` derive on iterator types?

Reply:
It caused the error with the `I` type parameter, suggesting to restrict it to the `I: ndarray::Dimension`

Reply:
Then put that bound on the structs. The iterators are useless without those bounds anyways.